risc0/zkvm/src/host/server/exec/executor.rs

```diff
@@ -304,11 +305,6 @@ impl<'a> ExecutorImpl<'a> {
                             self.segments.push(segment_ref);
                             self.split(Some(post_image.into()))?
                         }
-                        ExitCode::SessionLimit => {
-                            let segment_ref = callback(segment)?;
```

Comment:
I'm wondering how we can get away with not having this part of the callback? I wonder if we could just leave the ExitCode stuff alone for the moment and I can refactor the SegmentRef stuff and then later revisit ExitCode?

Reply:
I'm not sure I understand what you are pointing to. Are you raising a concern about dropping the `callback` call that occurs as this location prior to this PR?

One option is simply to close this PR and leave #1306 open. I don't think it affects any known use cases, so maybe that's the right call. What do you think?